resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesVolumeUtils.scala

```diff
@@ -105,4 +109,13 @@ private[spark] object KubernetesVolumeUtils {
       throw new NoSuchElementException(key + s" is required for $msg")
     }
   }
+
+  private def verifySize(size: Option[String]): Unit = {
+    size.foreach { v =>
+      if (v.forall(_.isDigit) && parseLong(v) < 1024) {
+        throw new IllegalArgumentException(
+          s"Volume size `$v` is smaller than 1KiB. Missing units?")
+      }
+    }
```

Comment:
Is 1KB the minimum size we can specify for volume?

Reply:
Should we mention in some place (any doc?) that the minimum volume size is 1KB?

Reply:
No~ In this case, this value is ignored. Instead, the created volume size is determined by underlying K8s configuration.

For example, 8GiB in some K8s system.

Reply:
Oh, you mean if `verifySize` fails, 8GB (or other default value) will be used?

Reply:
Of course, a user can make a mistake like 10240, but this PR aims to handle the most common cases which we are sure that this is a mistake.

Reply:
Oh, no~ This PR aims to fail the job from the beginning.

Reply:
What I meant is the the previous behavior is 8GiB (or other default value).

Reply:
Oh, you mean, previously once user specify a value like 1KB, 8GB (or other default value) will be used.

But now the PR makes it failed immediately.

Reply:
Yes, right~

Reply:
The previous behavior was not what the user's intention at all. And, the job fails after many executor losses as described in the PR description.

Reply:
Hmm, if the goal is to limit the value to be larger than the default value, why don't we check against it but a value like 1KB? What will happen if user specify 10KB, for example? It will be used or 8GB (default value) will be used?

Reply:
Spark don't know the default values. It's controlled by K8s system admin.
> Hmm, if the goal is to limit the value to be larger than the default value, why don't we check against it but a value like 1KB? 

Yes, if the user has a `unit` like KB, this PR will not block.
> What will happen if user specify 10KB, for example? It will be used or 8GB (default value) will be used?

And, as I mentioned in the above,
https://github.com/apache/spark/pull/45061#discussion_r1482183784, if a user uses 10240, we will not complain because it's too big as a mistake.